airbrakes/data_handling/logger.py

```diff
@@ -74,20 +76,22 @@ def stop(self) -> None:
     def log(self, logged_data_packets: collections.deque[LoggedDataPacket]) -> None:
         """
         Logs the current state, extension, and IMU data to the CSV file.
+
         :param logged_data_packets: the list of IMU data packets to log
         """
         # Loop through all the IMU data packets
         for logged_data_packet in logged_data_packets:
             # Formats the log message as a CSV line
-            message_dict = {key: getattr(logged_data_packet, key) for key in logged_data_packet.__struct_fields__}
+            # We are populating a dictionary with the fields of the logged data packet
+            message_dict = msgspec.structs.asdict(logged_data_packet)
```

Comment:
we should probably just import asdict, and then use that instead of accessing the attribute every iteration